en/guides/typesafesessions.md

```diff
@@ -0,0 +1,154 @@
+<div class="titleBlock">
+  <h1>Type-Safe Sessions</h1>
+  <p>How to add Session support to Kitura's Codable Routing</p>
+</div>
+
+## Introduction
+Kitura 2.4 introduces "Type-safe Middleware": middlewares that have a structure and data types that you define, that are instantiated and passed to your handler upon a request. Type-Safe Sessions is an implementation of `TypeSafeMiddleware` that allows an application developer to describe the exact structure and content of a Session, in the form of a `class` conforming to the `TypeSafeSession` protocol.
+
+The key feature is in the name: type safety. The previous implementation of Session that can be used alongside "Raw Routing" provides you with a general `[String: Any]` store, into which any type could be stored and retrieved. This places the burden of type checking on the user: any value retrieved from the session would need a downcast to its expected type, resulting in boilerplate checking and error handling, and the possibility of runtime errors.
```

Comment:
Why does session sometimes have a capital S and sometimes a lowercase s within this paragraph?